libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -521,9 +521,17 @@ static void _destroyMqttConnection( _mqttConnection_t * pMqttConnection )
     int8_t contextIndex = -1;
     bool mutexStatus = true;
 
+    IotMqtt_Assert( pMqttConnection != NULL );
+
     /* Getting MQTT Context for the specified MQTT Connection. */
     contextIndex = _IotMqtt_getContextIndexFromConnection( pMqttConnection );
-    IotMqtt_Assert( contextIndex != -1 );
+
+    /* An MQTT connection can be created without having MQTT Context associated.
+     * This can happen in case of an error in #IotMqtt_Connect() API. */
```

Comment:
How does the `_destroyMqttConnection` function get invoked if the `IotMqtt_Connect` API failed?

Reply:
Thanks for the clarification.